k9mail/src/main/java/com/fsck/k9/provider/UnreadWidgetProvider.java

```diff
@@ -47,7 +47,7 @@ public static void updateUnreadCount(Context context) {
     }
 
     public static void updateWidget(Context context, AppWidgetManager appWidgetManager,
-            int appWidgetId, String accountUuid) {
+            int appWidgetId, String accountUuid, String folderName) {
 
         RemoteViews remoteViews = new RemoteViews(context.getPackageName(),
```

Comment:
I like the structs idea, so I've implemented that. It contains the logic for calculating the clickIntent, unreadCount and title.